scripts/screeps_rl_experiment_card.py

```
@@ -273,6 +599,81 @@ def validate_strategy_id(value: str, label: str) -> None:
+    validate_strategy_id(raw["strategyVariantId"], f"policy_gradient.candidate_parameter_vectors[{index}].strategyVariantId")
+    validate_strategy_id(raw["sourceStrategyId"], f"policy_gradient.candidate_parameter_vectors[{index}].sourceStrategyId")
+    if raw["family"] != CONSTRUCTION_PRIORITY_FAMILY:
+        raise CardValidationError(f"policy_gradient.candidate_parameter_vectors[{index}].family must be construction-priority")
+    parameters = raw.get("parameters")
+    if not isinstance(parameters, dict):
+        raise CardValidationError(f"policy_gradient.candidate_parameter_vectors[{index}].parameters must be an object")
+    for knob in CONSTRUCTION_PRIORITY_KNOB_NAMES:
+        if knob not in parameters:
+            raise CardValidationError(
+                f"policy_gradient.candidate_parameter_vectors[{index}].parameters.{knob} is required"
+            )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)